plugins/plugins-api/src/main/java/org/codice/compliance/saml/plugin/IdpPostResponse.java

```diff
@@ -70,75 +72,94 @@ public IdpPostResponse build() {
     }
   }
 
-  // General
-  private int httpStatusCode;
+  private static final String VALUE = "value";
+  private static final String TYPE = "type";
+  private static final String NAME = "name";
+
   private Node samlResponseForm;
-  private String samlResponse;
-  private String relayState;
 
-  // Flags
+  private boolean isSamlResponseHidden;
   private boolean isRelayStateHidden;
-  private boolean isRelayStateGiven;
 
   @SuppressWarnings("squid:S3398" /* Method in here to simplify builder class */)
   private void parseAndSetFormValues(Node samlResponseForm) {
     this.samlResponseForm = samlResponseForm;
 
     // Bindings 3.5.4 "If the message is a SAML response, then the form control MUST be named
     // SAMLResponse."
-    samlResponse =
+    Node samlResponseNode =
         samlResponseForm
-            .getNodes("input")
+            .children()
+            .list()
             .stream()
-            .filter(node -> node.name().equals(SAML_RESPONSE))
-            .map(Node::value)
+            .filter(
+                node ->
+                    (node.attributes().get(NAME) != null
+                        && node.attributes().get(NAME).equals(SAML_RESPONSE)))
             .findFirst()
             .orElse(null);
 
+    // Bindings 3.5.4 "If a “RelayState” value is to accompany the SAML protocol message, it MUST be
+    // placed in an additional **hidden** form control named RelayState within the same form with
+    // the SAML message"
+
     Node relayStateNode =
         samlResponseForm
-            .getNodes("input")
+            .children()
+            .list()
             .stream()
-            .filter(node -> node.name().equals(RELAY_STATE))
+            .filter(
+                node ->
+                    (node.attributes().get(NAME) != null
+                        && node.attributes().get(NAME).equals(RELAY_STATE)))
             .findFirst()
             .orElse(null);
 
-    // Bindings 3.5.4 "If a “RelayState” value is to accompany the SAML protocol message, it MUST be
-    // placed in an additional hidden form control named RelayState within the same form with the
-    // SAML message"
-    if (relayStateNode != null) {
-      relayState = relayStateNode.value();
-      if (relayStateNode.getAttribute("type") != null) {
-        isRelayStateHidden = relayStateNode.getAttribute("type").equals("hidden");
+    /*
+     * Bindings 3.5.4 "A SAML protocol message is form-encoded by... placing the result **in** a
+     * **hidden** form control within a form as defined by [HTML401] Section 17"
+     *
+     * The two key words here are "in" and "hidden"
+     *
+     * Assuming "in" in the above quote means in either the value attribute or in the value
+     * itself.
+     *
+     * And "hidden" means both the SAMLResponse and RelayState MUST be placed in "hidden" form controls
+     */
+    // SAMLResponse portion
+    if (samlResponseNode != null) {
```

Comment:
Id prefer to keep the null checks, because then I would have to repeatedly call `relayStateNode.getValue` or assign it to a local variable. Those solutions just seem...messier.

Unless, of course, there is a good reason to do so from a best practices perspective.